server/container/util-java8/src/test/java/org/apache/james/util/streams/FakeSmtp.java

```diff
@@ -0,0 +1,54 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.util.streams;
+
+import java.net.Socket;
+
+import javax.net.SocketFactory;
+
+import com.jayway.awaitility.Awaitility;
+import com.jayway.awaitility.Duration;
+import com.jayway.awaitility.core.ConditionFactory;
+
+public class FakeSmtp extends SwarmGenericContainer {
```

Comment:
Can you use ``FakeSmtp`` everywhere you can use ``SwarmGenericContainer``.
I doubt no. And if no in this case you should not use inheritance but composition.
Actually reading more the class I think it's really weird to extend ``SwarmGenericContainer`` just to add a static helper.
I would have just created a helper class named something like ``ContainerWaitStrategy`` with the await method above with a extra argument ``port``.

After if you want to ease test, you could create a FakeSmtp class with composition over ``SwarmGenericContainer`` and with a non static method ``await`` that will call the static method discribed above with the 25 port